test/prif_test_s.f90

```diff
@@ -0,0 +1,76 @@
+! Copyright (c) 2020-2024, The Regents of the University of California
+! Terms of use are as specified in LICENSE.txt
+submodule(prif_test_m) prif_test_s
+  use iso_c_binding, only : c_funloc, c_bool
+  use julienne_m, only : command_line_t
+  use prif, only : prif_this_image_no_coarray, prif_co_broadcast, prif_co_reduce
+  use prif_test_m, only : test_description_substring
+  implicit none
+
+contains
+
+  impure elemental subroutine co_all(boolean)
+    logical(c_bool), intent(inout) :: boolean
+    call prif_co_reduce(boolean, c_funloc(both))
```

Comment:
This is using the PRIF 0.4 version of `prif_co_reduce`, and will break when we upgrade the implementation to PRIF 0.5, which significantly complicates the code needed to invoke `prif_co_reduce` (see PR #158).

Moreover there's no semantic reason to use a custom reduction function here. Please convert the boolean to an integer and use `prif_co_sum` or `prif_co_max` instead, to avoid the mess mentioned above.